CONTRIBUTING.rst

```diff
@@ -36,6 +39,47 @@ If modifying dependencies in ``mlflow/server/js/package.json``, run ``npm update
 ``mlflow/server/js`` to install the updated dependencies.
 
 
+Java API
+--------
+
+The MLflow Java client depends on the Python client, so first install the Python client in a conda
+environment as described above. `Install <https://www.oracle.com/technetwork/java/javase/downloads/index.html>`_
+the Java 8 JDK (or above), and `download <https://maven.apache.org/download.cgi>`_
+and `install <https://maven.apache.org/install.html>`_ Maven. You can then build and run tests via:
+
+.. code:: bash
+
+  cd mlflow/java
+  mvn clean package
```

Comment:
Probably we can remove the `clean` here. Perhaps a more explicit example, for people unfamiliar with maven terminology, would be `mvn compile test` (even though "compile" is not technically necessary, at "test" would otherwise first compile).